python/cudf/cudf/core/column/numerical_base.py

```diff
@@ -10,11 +10,12 @@
 
 import cudf
 from cudf import _lib as libcudf
-from cudf._typing import Dtype, ScalarLike
+from cudf._typing import ScalarLike
 from cudf.core.column import ColumnBase
+from cudf.core.mixins import Reducible
 
 
-class NumericalBaseColumn(ColumnBase):
+class NumericalBaseColumn(ColumnBase, Reducible):
```

Comment:
`ColumnBase` already inherits from Reducible, why do we need to do it again here?

Reply:
From this example, `__init_sublcass__` seems rerun every time you initialize a new subclass:
```python
In [22]: class Mixin:
    ...:     @classmethod
    ...:     def __init_subclass__(cls):
    ...:         print(f"initting sublclass {cls.__name__}!")
    ...: 

In [23]: class A(Mixin):
    ...:     pass
    ...: 
initting sublclass A!

In [24]: class B(A):
    ...:     pass
    ...: 
initting sublclass B!
```

Reply:
You're right that we don't have to, I just did it to make it explicit. But yeah, we can remove this.